diff -Nur camb_Nov13_orig/modules.f90 camb_Nov13/modules.f90
--- camb_Nov13_orig/modules.f90	2014-05-10 17:59:12.000000000 -0500
+++ camb_Nov13/modules.f90	2014-05-10 23:15:59.000000000 -0500
@@ -58,7 +58,7 @@
     !Nu_best: automatically use mixture which is fastest and most accurate
 
     integer, parameter :: max_Nu = 5 !Maximum number of neutrino species
-    integer, parameter :: max_transfer_redshifts = 150
+    integer, parameter :: max_transfer_redshifts = 500 ! COSMOSIS - alter number of transfer redshifts
     integer, parameter :: fileio_unit = 13 !Any number not used elsewhere will do
     integer, parameter :: outNone=1
 
