diff -Nur camb_Nov13_orig/modules.f90 camb_Nov13/modules.f90
--- camb_Nov13_orig/modules.f90	2014-05-10 17:59:12.000000000 -0500
+++ camb_Nov13/modules.f90	2014-05-10 19:48:16.000000000 -0500
@@ -1010,7 +1010,9 @@
     deallocate(CTrans%Delta_p_l_k, STAT = st)
     call Ranges_getArray(CTrans%q, .true.)
 
-    allocate(CTrans%Delta_p_l_k(CTrans%NumSources,min(CTrans%max_index_nonlimber,CTrans%ls%l0), CTrans%q%npoints))
+    ! COSMOSIS - allocate up to max_bessels_l_index rather than CTrans%max_index_nonlimber
+    !allocate(CTrans%Delta_p_l_k(CTrans%NumSources,min(CTrans%max_index_nonlimber,CTrans%ls%l0), CTrans%q%npoints))
+    allocate(CTrans%Delta_p_l_k(CTrans%NumSources,min(max_bessels_l_index,CTrans%ls%l0), CTrans%q%npoints))
     CTrans%Delta_p_l_k = 0
 
     end subroutine Init_ClTransfer
